### ISSUE TITLE
Problem about the order and the unet result

### ISSUE THREAD

**Issue by taotianli** [mabhijithn/irregulars-neureka-codebase#1]
After reading your article about the Neureka Challenge 2020, I would like to ask some questions about the code of seizure detection algorithm.  

1. Prediction results of Unet. After I ran the file (irregulars-neureka-codebase/evaluate/2-unet-prediction.py), I found that the length of the ground truth and the predicted result were not the same. I want to know what caused the issue. 

2. Run file order. After I have trained Unet (/train/3-train-unet), should I execute unet prediction (/evaluate/2-unet-prediction) next to get prediction results ? And then I use this result to train the LSTM, right?

3. The difference between the function - build_unet() and build_windowfree_unet(). I found that you used different Unet functions in the training and evaluation. Why do you use Window free? Can I use the function build_unet() to make a prediction?

Best，looking forward to your reply，

**Comment by nseeuws** [comment]
1. Prediction results
We drop some time samples at the end of an EEG recording to ensure the recording can be processed by the CNN in a single pass. In the worst case this can ignore the final five seconds of the signal. For the Neureka challenge, we assumed those five seconds wouldn’t matter much, but for your application this might be a problem. It should be possible to merge several predictions to ensure a final prediction with the same length as your ground truth if need be.

2. Run file order
Correct. First, you train a U-net and use it to generate predictions (you can think of this as a signal of predictions). The LSTM is then trained using multiple of these U-net predictions (all trained on different “views” of your data) as input to predict the final seizure labels (you can think of the LSTM looking for the optimal way of merging multiple prediction signals)

3. Different U-nets
The U-net is trained using signal segments of fixed size, while the prediction files process EEG signals of varying duration. The two functions are used so we can do this within the Keras framework. build_unet() gives you a network that can also be used for prediction, but you would have to provide EEG signals with exactly 4096 time samples. The two architectures are identical, they can just process inputs of different sizes. (You can use both for training and for prediction, but using a fixed size for training makes it easier to implement and, as far as I know, run faster)
